packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -138,13 +205,38 @@ export class Fleet extends Resource implements IFleet {
       throw new Error('baseCapacity must be greater than or equal to 1');
     }
 
-    super(scope, id, { physicalName: props.fleetName });
+    if (
+      props.computeType === FleetComputeType.ATTRIBUTE_BASED &&
+      (!props.computeConfiguration || Object.keys(props.computeConfiguration).length === 0)
+    ) {
+      throw new Error('At least one compute configuration criteria must be specified if computeType is "ATTRIBUTE_BASED"');
+    }
+    if (props.computeConfiguration && props.computeType !== FleetComputeType.ATTRIBUTE_BASED) {
+      throw new Error(`'computeConfiguration' can only be specified if 'computeType' is 'ATTRIBUTE_BASED', got: ${props.computeType}`);
+    }
+
+    const computeConfig = props.computeConfiguration;
+    const diskGiB = computeConfig?.disk?.toGibibytes() ?? 0;
+    const memoryGiB = computeConfig?.memory?.toGibibytes() ?? 0;
+    const vCpu = computeConfig?.vCpu ?? 0;
```

Comment:
Probably a good idea to add a comment to explain why we need to have these default values here as well

Reply:
Thanks. Your PR had an excellent comment, so I used it as it was.

```ts
    // Despite what the CloudFormation schema says, the numeric properties are not optional.
    // An undefined value will cause the CloudFormation deployment to fail, e.g.
    // > Cannot invoke "java.lang.Integer.intValue()" because the return value of "software.amazon.codebuild.fleet.ComputeConfiguration.getMemory()" is null
```